osf/metadata/serializers.py

```diff
@@ -0,0 +1,136 @@
+import json
+from datacite import schema40
+
+from osf.metadata import utils
+from website.settings import DOMAIN
+
+serializer_registry = {}
+
+def register(schema_id):
+    """Register classes into serializer_registry"""
+    def decorator(cls):
+        serializer_registry[schema_id] = cls
+        return cls
+    return decorator
+
+
+class MetadataRecordSerializer(object):
+
+    def serialize_json(self, metadata_record):
+        raise NotImplementedError
+
+    def serialize_xml(self, metadata_record):
+        raise NotImplementedError
+
+    @classmethod
+    def serialize(cls, metadata_record, format='json'):
+        if format == 'json':
+            return cls.serialize_json(metadata_record)
+        if format == 'xml':
+            return cls.serialize_xml(metadata_record)
+        raise ValueError('Format "{}" is not supported.'.format(format))
+
+
+@register(schema_id='datacite')
+class DataciteMetadataRecordSerializer(MetadataRecordSerializer):
+
+    osf_schema = 'osf_datacite.json'
+
+    @classmethod
+    def serialize_json(cls, record):
+        osfstorage_file = record.file
+        target = osfstorage_file.target
+        doc = {
+            'creators': utils.datacite_format_contributors(target.visible_contributors),
+            'titles': [
+                {
+                    'title': osfstorage_file.name
+                },
+                {
+                    'title': target.title,
+                    'titleType': 'AlternativeTitle'
+                }
+            ],
+            'publisher': 'Open Science Framework',
```

Comment:
What about serializing preprint file metadata for a branded preprint?  Trying to serialize a psyarxiv preprint file, for example.

Reply:
This is a great question! I am getting a strong sense of deja-vu from this exact question when minting preprint DOIs with CrossRef and what to put for the "publisher" and OSF being decided on no matter the branded preprint -- I think the reasoning was similar. I will track down documentation/an official stance on this tho!

Reply:
Updating here for clarity -- discussed in a meeting, leaving it this way so that it matches CrossRef's metadata for preprints.